NeosModLoader/Utility/GenericTypeMethodsInvoker.cs

```diff
@@ -0,0 +1,166 @@
+		{ }
+
+		public GenericTypeMethodsInvoker(Type genericType, Func<MethodInfo, Type, MethodInfo> getGenericMethodOfConcreteType)
+		{
+			GenericType = genericType;
+			GetGenericMethodOfConcreteType = getGenericMethodOfConcreteType;
+		}
+
+		public TReturn Invoke<TReturn>(MethodInfo method, Type[] instanceTypes, object instance, Type[] methodTypes, params object[] parameters)
+		{
+			return (TReturn)InvokeInternal(method, instanceTypes, instance, methodTypes, parameters);
+		}
+
+		public TReturn Invoke<TReturn>(MethodInfo method, Type[] instanceTypes, object instance, Type methodType, params object[] parameters)
+		{
+			return (TReturn)InvokeInternal(method, instanceTypes, instance, methodType, parameters);
+		}
+
+		public object Invoke(MethodInfo method, Type[] instanceTypes, object instance, Type[] methodTypes, params object[] parameters)
+		{
+			return InvokeInternal(method, instanceTypes, instance, methodTypes, parameters);
+		}
+
+		public object Invoke(MethodInfo method, Type instanceType, object instance, Type[] methodTypes, params object[] parameters)
+		{
+			return InvokeInternal(method, instanceType, instance, methodTypes, parameters);
+		}
+
+		public object Invoke(MethodInfo method, Type[] instanceTypes, object instance, Type methodType, params object[] parameters)
+		{
+			return InvokeInternal(method, instanceTypes, instance, methodType, parameters);
+		}
+
+		public object Invoke(MethodInfo method, Type instanceType, object instance, Type methodType, params object[] parameters)
+		{
+			return InvokeInternal(method, instanceType, instance, methodType, parameters);
+		}
+
+		public TReturn Invoke<TReturn>(MethodInfo method, Type instanceType, object instance, Type[] methodTypes, params object[] parameters)
+		{
+			return (TReturn)InvokeInternal(method, instanceType, instance, methodTypes, parameters);
+		}
+
+		public TReturn Invoke<TReturn>(MethodInfo method, Type instanceType, object instance, Type methodType, params object[] parameters)
+		{
+			return (TReturn)InvokeInternal(method, instanceType, instance, methodType, parameters);
+		}
+
+		public TReturn Invoke<TReturn>(MethodInfo method, Type instanceType, object instance, params object[] parameters)
+		{
+			return (TReturn)InvokeInternal(method, instanceType, instance, new TypeDefinition(), parameters);
+		}
+
+		public TReturn Invoke<TReturn>(MethodInfo method, Type instanceType, params object[] parameters)
+		{
+			return (TReturn)InvokeInternal(method, instanceType, null, new TypeDefinition(), parameters);
+		}
+
+		public object Invoke(MethodInfo method, Type[] instanceTypes, object instance, params object[] parameters)
+		{
+			return InvokeInternal(method, instanceTypes, instance, new TypeDefinition(), parameters);
+		}
+
+		public object Invoke(MethodInfo method, Type[] instanceTypes, params object[] parameters)
+		{
+			return InvokeInternal(method, instanceTypes, null, new TypeDefinition(), parameters);
+		}
```

Comment:
Similarly as the previous comment these 4, couldn't the `instance` object just be nullable..?

Reply:
they're for convenience so that you don't need to pass null when invoking a static method, but if you make it an optional parameter as `object? instance = null` with the `params object[] parameters` behind it, it will create an ambiguous match if you have an instance and no parameters or no instance and one parameter